src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/ResolutionTransactionMapper.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.companieshouse.filinghistory.api.mapper.upsert;
+
+import static uk.gov.companieshouse.filinghistory.api.mapper.DateUtils.stringToInstant;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.filinghistory.InternalData;
+import uk.gov.companieshouse.api.filinghistory.InternalFilingHistoryApi;
+import uk.gov.companieshouse.filinghistory.api.exception.ConflictException;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryData;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryDocument;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryResolution;
+
+@Component
+public class ResolutionTransactionMapper extends AbstractTransactionMapper {
+
+    private final TopLevelTransactionMapper topLevelTransactionMapper;
+    private final ChildMapper<FilingHistoryResolution> resolutionChildMapper;
+    private final Supplier<Instant> instantSupplier;
+
+    public ResolutionTransactionMapper(LinksMapper linksMapper, TopLevelTransactionMapper topLevelTransactionMapper,
+            ChildMapper<FilingHistoryResolution> resolutionChildMapper, Supplier<Instant> instantSupplier) {
+        super(linksMapper);
+        this.topLevelTransactionMapper = topLevelTransactionMapper;
+        this.resolutionChildMapper = resolutionChildMapper;
+        this.instantSupplier = instantSupplier;
+    }
+
+    @Override
+    protected FilingHistoryData mapFilingHistoryData(InternalFilingHistoryApi request, FilingHistoryData data) {
+        topLevelTransactionMapper.mapFilingHistoryData(request, data);
+        return data.resolutions(List.of(resolutionChildMapper.mapChild(new FilingHistoryResolution(), request)));
+    }
+
+    @Override
+    public FilingHistoryDocument mapFilingHistoryToExistingDocumentUnlessStale(InternalFilingHistoryApi request,
+            FilingHistoryDocument existingDocument) {
+
+        final String requestEntityId = request.getInternalData().getEntityId();
+
+        Optional.ofNullable(existingDocument.getData().getResolutions())
+                .ifPresentOrElse(
+                        resolutionList -> resolutionList.stream()
+                                .filter(resolution -> resolution.getEntityId().equals(requestEntityId))
+                                .findFirst()
+                                .ifPresentOrElse(resolution -> {
+                                            if (isDeltaStale(request.getInternalData().getDeltaAt(),
+                                                    resolution.getDeltaAt())) {
+                                                throw new ConflictException(
+                                                        "Delta at stale when updating annotation");
```

Comment:
Can you add the `entityId` to the exception message. It will help FES in their fault diagnosis if it appears in the logs in Live

Reply:
done, put it in Controller log message.